sdk/python/setup.py

```diff
@@ -78,7 +78,8 @@
 ]
 
 REDIS_REQUIRED = [
-    "redis>=4.1.0",
+    "redis>=3.5.3",
```

Comment:
If we're downgrading `redis`, should we pin to a max allowed version?

Reply:
I believe redis-py-cluster ensures that, should I still pin the version, this is autogenerated because redis-py-cluster requires something before redis-py 4.

Reply:
I think we should still pin the version since we're directly relying on that pin in our logic.